tests/bulk/bulkwrite-ctor-let_error-001.phpt

```diff
@@ -0,0 +1,18 @@
+--TEST--
+MongoDB\Driver\BulkWrite::__construct(): let option
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/basic.inc';
+
+echo throws(function() {
```

Comment:
May be it would be more safe to add a various of incorrect data tests, for example NULL value, which sometime produces borderline cases?

Reply:
This is far less likely to happen in the extension since we don't use things like `isset()` and examine the zval type directly. For example:

```c
if (Z_TYPE_P(value) != IS_OBJECT && Z_TYPE_P(value) != IS_ARRAY) {
```

That said, there's no reason we can't test more invalid types. In older PHP versions, some types were reported inconsistently (e.g. "integer" vs. "int"), but I don't think that's an issue with our current version requirement of PHP 7.2+.

I added some additional test cases.